src/Support/SupportData.php

```diff
@@ -0,0 +1,930 @@
+		}
+
+		if ( ! empty( $theme_object->parent() ) && ! is_a( $theme_object->parent(), 'WP_Theme' ) ) {
+			$parent_theme = $theme_object->parent()->get_stylesheet();
+		}
+
+		$tags = $theme_object->get( 'Tags' );
+		$tags = ( ! empty( $tags ) && is_array( $tags ) ) ? $tags : [];
+
+		$theme_data = [
+			'name'         => $theme_object->get( 'Name' ),
+			'slug'         => $theme_object->get_stylesheet(),
+			'version'      => $theme_object->get( 'Version' ),
+			'status'       => $theme_object->get( 'Status' ),
+			'tags'         => $tags,
+			'text_domain'  => $theme_object->get( 'TextDomain' ),
+			'requires_wp'  => $theme_object->get( 'RequiresWP' ),
+			'requires_php' => $theme_object->get( 'RequiresPHP' ),
+			'theme_url'    => $theme_object->get( 'ThemeURI' ),
+			'author'       => $theme_object->get( 'Author' ),
+			'author_url'   => $theme_object->get( 'AuthorURI' ),
+			'is_active'    => ( $theme_object->get_stylesheet() === $active_theme_slug ),
+			'parent_theme' => $parent_theme,
+		];
+
+		return $theme_data;
+	}
+
+	/**
+	 * Normalize error data.
+	 *
+	 * @since 2.2
+	 *
+	 * @param array $error_data Error data array.
+	 *
+	 * @return array
+	 */
+	public static function normalize_error( $error_data ) {
+
+		if ( empty( $error_data ) || ! is_array( $error_data ) ) {
+			return [];
+		}
+
+		unset( $error_data['sources'] );
+
+		$error_data['text'] = ( ! empty( $error_data['text'] ) ) ? trim( $error_data['text'] ) : '';
+
+		$error_data = wp_json_encode( $error_data );
+		$error_data = static::remove_domain( $error_data );
+		$error_data = json_decode( $error_data, true );
+
+		ksort( $error_data );
+
+		/**
+		 * Generate new slug after removing site specific data.
+		 */
+		$error_data['error_slug'] = static::generate_hash( $error_data );
+
+		return $error_data;
+	}
+
+	/**
+	 * To normalize the error source data.
+	 *
+	 * @since 2.2
+	 *
+	 * @param array $source Error source detail.
+	 *
+	 * @return array Normalized error source data.
+	 */
+	public static function normalize_error_source( $source ) {
+
+		if ( empty( $source ) || ! is_array( $source ) ) {
+			return [];
+		}
+
+		static $plugin_versions = [];
+		static $theme_versions  = [];
+
+		/**
+		 * All plugin info
+		 */
+		if ( empty( $plugin_versions ) || ! is_array( $plugin_versions ) ) {
+
+			$plugin_list = get_plugins();
+			$plugin_list = array_keys( $plugin_list );
+			$plugin_list = array_values( array_unique( $plugin_list ) );
+			$plugin_list = array_map( __CLASS__ . '::normalize_plugin_info', $plugin_list );
+
+			foreach ( $plugin_list as $plugin ) {
+				$plugin_versions[ $plugin['slug'] ] = $plugin['version'];
+			}
+		}
+
+		/**
+		 * All theme info.
+		 */
+		if ( empty( $theme_versions ) || ! is_array( $theme_versions ) ) {
+
+			$theme_list = array_merge( [ wp_get_theme() ], wp_get_themes() );
+
+			foreach ( $theme_list as $theme ) {
+				if ( ! empty( $theme ) && is_a( $theme, 'WP_Theme' ) ) {
```

Comment:
This condition is not needed because `wp_get_theme()` and `wp_get_themes()` _always_ return `WP_Theme` objects.

Reply:
If wanting to check for errors, this should rather be:

```suggestion
				if ( ! $theme->errors() ) {
```